utils/epinow2/functions.py

```diff
@@ -9,6 +10,29 @@
 )
 
 
+def extract_user_args() -> dict:
+    """Extracts user-provided arguments from environment variables or uses default if none provided."""
+    state = os.environ.get("state", "all")
+    disease = os.environ.get("disease", "all")
+    report_date = os.environ.get("report_date", date.today())
+    min_reference_date, max_reference_date = get_reference_date_range(report_date)
+    reference_dates = os.environ.get(
+        "reference_date", [min_reference_date, max_reference_date]
+    )
```

Comment:
In this case, `reference_dates` is either a single `str` if read directly from the environment, or a `list[date]` for the case nothing was passed in. In the case of the single `str`, how should the user set as the environment variable?
1. `reference_dates="YYYY-MM-DD,YYYY-MM-DD"`
1. `reference_dates="YYYY-MM-DD YYYY-MM-DD"`
1. something else?

Personally, I would vote for option 1, but I don't have a strong feeling there. Acu

Whatever we choose, I think we need to split it and parse it here, or at least split it, and leave the parsing to `validate_args()`.

Reply:
@natemcintosh my initial thought was that `reference_date` would always be a list. Either we use the default `list[date]` or it'll get passed in as a list of strings `list[str]` in iso format. But now that you mention it, it seems that GHA doesn't support this, so let's go with your option 1. Will implement and update the validate fn